tests/progs/test_mimc.py

```diff
@@ -20,13 +20,35 @@
 
         # Compute F_MiMC_mpc
         mm = await mimc_mpc(context, x, key)
+        mm_open = await mm.open()
 
         # open x, then compute F_MiMC_plain
         x_open = await x.open()
         mp = mimc_plain(x_open, key)
 
         # Compare the MPC evaluation to the plain one
-        mm_ = await mm.open()
-        assert mm_ == mp
+        assert mm_open == mp
+
+    await test_runner(_prog, n, t, PREPROCESSING, k, MIXINS)
+
+
+@mark.asyncio
+async def test_mimc_mpc_batch(test_preprocessing, test_runner):
+
+    async def _prog(context):
+        xs = [test_preprocessing.elements.get_rand(context) for _ in range(20)]
+        field = GF(Subgroup.BLS12_381)
+        key = field(15)
```

Comment:
can you pick key as a random element here rather than 15?

Reply:
Sure.